File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EPrimaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EPrimaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd16
            AppTheme.STANDARD -> R.color.cstd16
            AppTheme.BLUE -> R.color.cbd16
            AppTheme.GREEN -> R.color.cgd16
            AppTheme.RED -> R.color.crd16
            AppTheme.NOTHING -> R.color.cnod16
        }
    }
}

File: lib/layer2_core/dark/e_primary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EPrimaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd16;
      case 'standard': return cstd16;
      case 'blue': return cbd16;
      case 'green': return cgd16;
      case 'red': return crd16;
      case 'nothing': return cnod16;
      default: return cstd16;
    }
  }
}
